### DISCUSSION TITLE
Is it time for a Pydantic V2 tag on StackOverflow?

### DISCUSSION THREAD

**Discussion by ybressler** [pydantic/pydantic#6705] [General]
When people ask their Pydantic Q's on StackOverflow, they have the option to [tag them `pydantic`](https://stackoverflow.com/questions/tagged/pydantic). Currently, lots of the Q's are specific to Pydantic-V1. The answers are even more specific to V1.

As V2 becomes adopted, these Q's and their answers will get in the way of people looking for help. I'd like to suggest we change the github tags – here are a few solutions:

| V1 Tag  | V2 Tag    | Description |
| :-------- | :-------------- | :------- |
| `pydantic-v1` | `pydantic`  | Change all `pydantic` tags to `pydantic-v1`. All new Q's use `pydantic`   |
| `pydantic-v1` | `pydantic-v2`  | Change all `pydantic` tags to `pydantic-V1` + create a new tag `pydantic-v2`   |
| `pydantic` | `pydantic-v2`  | Create a new tag `pydantic-v2`, leave old tags alone.   |

**Recommendation:** I personally like choice 1 the most, since it allows SO contributors to determine if a Q is for V1, but default as V2. Onus is on contributor.

> **Some background:** There are only [1,520 Q's tagged as `pydantic`](https://stackoverflow.com/questions/tagged/pydantic) – so overhead of migration would be minimal.

**Side Note:** I'd be curious how @tiangolo is approaching this with FastAPI...

**Comment by samuelcolvin** [comment]
> github tags

Do you mean StackOverflow  tags?

If so, I don't think we can change them easily, and even creating new tags is hard.

Personally I think sticking to a single tag shouldn't be a problem in the long term.

If people understand the distinction between the versions, they will put it in the question body, if they don't, there's no hope either way.

**Reply by ybressler** [reply]
I have SO privileges of creating new tags. I also have privileges to migrate tags. (I am a top contributor.)

**Reply by samuelcolvin** [reply]
Nice.

What do other projects like Django or tensorflow do?

**Reply by ybressler** [reply]
Lemme go take a look and see. (Also, kudos to you for super speedy responses.)

**Reply by ybressler** [reply]
Looks like things trend towards entropy:

<img width="300" alt="Screenshot 2023-07-16 at 3 29 16 PM" src="https://github.com/pydantic/pydantic/assets/40807730/642cb447-c8ce-431f-91a4-ece6dddc6428">
<img width="450" alt="Screenshot 2023-07-16 at 3 29 31 PM" src="https://github.com/pydantic/pydantic/assets/40807730/9184d456-a0fa-4411-849b-0a8102b981bc">

**Reply by ybressler** [reply]
Tensorflow:
* https://stackoverflow.com/questions/tagged/tensorflow2.0

**Reply by ybressler** [reply]
Here's my concern – I want V1 content to basically be deprecated. So converting all the current Q's (manually) to `pydantic-v1` will allow all future Q's to be cleaner.

**Reply by ybressler** [reply]
For short A's, people can mention both versions: https://stackoverflow.com/a/59126638/10521959

But as the answer gets complex, this will quickly get out of hand.

**Reply by ybressler** [reply]
I want to avoid stuff like this: https://stackoverflow.com/a/50852627/10521959

**Reply by ybressler** [reply]
I also want to avoid this too: https://stackoverflow.com/questions/34097281/convert-a-tensor-to-numpy-array-in-tensorflow

Unless you think that's necessary >> old Q's begin to have multiple answers depending on Pydantic V...

**Comment by ybressler** [comment]
Also, we can edit this wiki to provide more info: https://stackoverflow.com/tags/pydantic/info
